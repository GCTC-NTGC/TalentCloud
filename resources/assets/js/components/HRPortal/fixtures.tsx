import { text, boolean, number } from "@storybook/addon-knobs";
import { JobAction } from "../../models/types";
import { JobStatus } from "../../models/lookupConstants";

export const jobActions: JobAction[] = [
  {
    activity: {
      count: 3,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "CS01",
    draft: {
      url: "https://google.ca",
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: true,
    preview: {
      url: "https://google.ca",
      text: "Preview Poster",
      title: "Preview what this Job Poster will look like.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Draft,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Front-end Developer",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 12,
    classification: "BK01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Published,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Cupcake Baker",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "BS01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Complete,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Underwater Basket Weaver",
    userTime: 2,
  },
];

export const unclaimedJobs = [
  {
    title: "CS01 - Front-end Developer",
    createdAt: "2019-MAY-02",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: [],
  },
  {
    title: "AS02 - Executive Assisstant",
    createdAt: "2019-MAR-12",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: ["Rebecca Appleby", "Jack Little"],
  },
  {
    title: "ET03 - Business Analyst",
    createdAt: "2019-DEC-02",
    status: JobStatus.Review,
    hiringManagers: ["Robin Browne"],
    hrAdvisors: [],
  },
  {
    title: "FG05 - Long-term Care Specialist",
    createdAt: "2019-MAY-14",
    status: JobStatus.Draft,
    hiringManagers: ["Braeden McDoogal"],
    hrAdvisors: ["Caitlyn Summers", "Jack Little"],
  },
  {
    title: "CS03 - Digital Product Designer",
    createdAt: "2019-JUL-24",
    status: JobStatus.Review,
    hiringManagers: ["Amelie Lachance"],
    hrAdvisors: [],
  },
];
