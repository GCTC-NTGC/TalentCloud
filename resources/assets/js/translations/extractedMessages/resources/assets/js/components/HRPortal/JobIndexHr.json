[
  {
    "id": "hrPortal.jobPageIndex.completedJobsHeader",
    "description": "Header for completed jobs accordion section.",
    "defaultMessage": "My Completed Job Actions"
  },
  {
    "id": "hrPortal.jobPageIndex.clickToView",
    "description": "Accordion trigger message for screen readers.",
    "defaultMessage": "Click to view..."
  },
  {
    "id": "hrPortal.jobPageIndex.showAccordion",
    "description": "Accordion trigger message to show items.",
    "defaultMessage": "Show"
  },
  {
    "id": "hrPortal.jobPageIndex.hideAccordion",
    "description": "Accordion trigger message to hide items.",
    "defaultMessage": "Hide"
  },
  {
    "id": "hrPortal.jobPageIndex.noJobsCompleted",
    "description": "Message displayed if the completed jobs list is empty.",
    "defaultMessage": "No jobs completed yet!"
  },
  {
    "id": "hrPortal.jobPageIndex.welcomeMessage",
    "description": "Welcome message at beginning of page.",
    "defaultMessage": "Welcome! Introductory copy that explains how this page works, and what an HR advisor needs to do to claim a job action as their own. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laudantium ducimus laboriosam sequi, quis autem minima esse quasi aspernatur vero provident quos eligendi, ea officia exercitationem. Obcaecati impedit quae veritatis corrupti!"
  },
  {
    "id": "hrPortal.jobPageIndex.jobActionsHeader",
    "description": "Header for my job actions section.",
    "defaultMessage": "My Job Actions"
  },
  {
    "id": "hrPortal.jobPageIndex.jobActionsMessage",
    "description": "Message before list of users job actions.",
    "defaultMessage": "This is a list of all job actions you are currently participating in. Looking for an older job? Check the 'My Completed Job Actions' section below your active jobs."
  },
  {
    "id": "hrPortal.jobPageIndex.jobActionsEmpty",
    "description": "Message displayed if the jobs actions list is empty.",
    "defaultMessage": "Claim a job below!"
  },
  {
    "id": "hrPortal.jobPageIndex.preDepartmentName",
    "description": "Message before department name.",
    "defaultMessage": "All Jobs in"
  },
  {
    "id": "hrPortal.jobPageIndex.unclaimedJobsMessage",
    "description": "Message before list of unclaimed jobs.",
    "defaultMessage": "This is the list of all active job actions in your department. From here you can 'claim' a job, which will move it into your jobs list above and allow you to begin working with the hiring manager on finding the best talent possible. If you claim a job by accident, no fear, for you can click into the job summary and remove yourself using the 'Relinquish This Job' button."
  },
  {
    "id": "hrPortal.jobPageIndex.unclaimedJobsEmpty",
    "description": "Message displayed if the unclaimed jobs list is empty.",
    "defaultMessage": "There are currently no active jobs available."
  }
]
