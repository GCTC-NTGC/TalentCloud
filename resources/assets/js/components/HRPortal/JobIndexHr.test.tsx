import React from "react";
import { shallow } from "enzyme";
import JobIndexHr from "./JobIndexHr";
import { jobActions, unclaimedJobs } from "./fixtures";

describe("Job Index Page", (): void => {
  it("should render JobIndexHr correctly", (): void => {
    const wrapper = shallow(
      <JobIndexHr
        jobActions={jobActions}
        unclaimedJobs={unclaimedJobs}
        departmentName="Treasury Board of Canada Secretariat"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render JobIndexHr without any items", (): void => {
    const wrapper = shallow(
      <JobIndexHr
        jobActions={[]}
        unclaimedJobs={[]}
        departmentName="Treasury Board of Canada Secretariat"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
