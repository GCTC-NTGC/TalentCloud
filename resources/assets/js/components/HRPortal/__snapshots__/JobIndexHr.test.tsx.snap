// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Index Page should render JobIndexHr correctly 1`] = `
<section>
  <div
    data-c-background="gray(10)"
    data-c-border="bottom(thin, solid, black)"
  >
    <div
      data-c-container="large"
      data-c-padding="tb(triple)"
    >
      <h2
        data-c-font-size="h2"
        data-c-margin="bottom(normal)"
      >
        <FormattedMessage
          defaultMessage="My Job Actions"
          description="Header for my job actions section."
          id="hrPortal.jobPageIndex.jobActionsHeader"
        />
      </h2>
      <p
        data-c-margin="bottom(double)"
      >
        <FormattedMessage
          defaultMessage="This is a list of all job actions you are currently participating in. Looking for an older job? Check the 'My Completed Job Actions' section below your active jobs."
          description="Message before list of users job actions."
          id="hrPortal.jobPageIndex.jobActionsMessage"
        />
      </p>
      <JobCard
        activity={
          Object {
            "count": 3,
            "new": Object {
              "text": "New Activity",
              "title": "Click here to see new activity on this posting.",
              "url": "https://google.ca",
            },
          }
        }
        applicants={
          Object {
            "text": "5 Applicants",
            "title": "",
            "url": "https://google.ca",
          }
        }
        classification="CS01"
        draft={
          Object {
            "text": "View Draft",
            "title": "View this draft Job Poster.",
            "url": "https://google.ca",
          }
        }
        id={1}
        key="1"
        managerTime={8}
        owned={true}
        preview={
          Object {
            "text": "Preview Poster",
            "title": "Preview what this Job Poster will look like.",
            "url": "https://google.ca",
          }
        }
        screeningPlan={
          Object {
            "text": "View Screening Plan",
            "title": "View the screening plan for this Job Poster.",
            "url": "https://google.ca",
          }
        }
        status={
          Object {
            "description": Object {
              "en": "This is a draft.",
              "fr": "This is a draft (FR).",
            },
            "id": 1,
            "key": "draft",
            "name": Object {
              "en": "Draft",
              "fr": "Provisoire",
            },
          }
        }
        summary={
          Object {
            "text": "Visit Job Summary",
            "title": "View the Job Summary for this Job Poster.",
            "url": "https://google.ca",
          }
        }
        title="Front-end Developer"
        userTime={2}
      />
      <JobCard
        activity={
          Object {
            "count": 0,
            "new": Object {
              "text": "New Activity",
              "title": "Click here to see new activity on this posting.",
              "url": "https://google.ca",
            },
          }
        }
        applicants={
          Object {
            "text": "12 Applicants",
            "title": "",
            "url": "https://google.ca",
          }
        }
        classification="BK01"
        draft={
          Object {
            "text": "View Draft",
            "title": "View this draft Job Poster.",
            "url": "",
          }
        }
        id={2}
        key="2"
        managerTime={8}
        owned={false}
        preview={
          Object {
            "text": "View Poster",
            "title": "View the live Job Poster.",
            "url": "https://google.ca",
          }
        }
        screeningPlan={
          Object {
            "text": "View Screening Plan",
            "title": "View the screening plan for this Job Poster.",
            "url": "https://google.ca",
          }
        }
        status={
          Object {
            "description": Object {
              "en": "This job has has gone live, and will accept applications between its open and closing dates.",
              "fr": "This job has has gone live, and will accept applications between its open and closing dates. (FR).",
            },
            "id": 9,
            "key": "live",
            "name": Object {
              "en": "Live",
              "fr": "En ligne",
            },
          }
        }
        summary={
          Object {
            "text": "Visit Job Summary",
            "title": "View the Job Summary for this Job Poster.",
            "url": "https://google.ca",
          }
        }
        title="Cupcake Baker"
        userTime={2}
      />
      <CompletedJobsAccordion
        completedJobActions={
          Array [
            Object {
              "activity": Object {
                "count": 0,
                "new": Object {
                  "text": "New Activity",
                  "title": "Click here to see new activity on this posting.",
                  "url": "https://google.ca",
                },
              },
              "applicants": Object {
                "text": "5 Applicants",
                "title": "",
                "url": "https://google.ca",
              },
              "classification": "BS01",
              "draft": Object {
                "text": "View Draft",
                "title": "View this draft Job Poster.",
                "url": "",
              },
              "id": 3,
              "managerTime": 8,
              "owned": false,
              "preview": Object {
                "text": "View Poster",
                "title": "View the published Job Poster.",
                "url": "https://google.ca",
              },
              "screeningPlan": Object {
                "text": "View Screening Plan",
                "title": "View the screening plan for this Job Poster.",
                "url": "https://google.ca",
              },
              "status": Object {
                "description": Object {
                  "en": "The process of this job poster has been completed.",
                  "fr": "The process of this job poster has been completed (FR).",
                },
                "id": 10,
                "key": "completed",
                "name": Object {
                  "en": "Completed",
                  "fr": "TerminÃ©",
                },
              },
              "summary": Object {
                "text": "Visit Job Summary",
                "title": "View the Job Summary for this Job Poster.",
                "url": "https://google.ca",
              },
              "title": "Underwater Basket Weaver",
              "userTime": 2,
            },
          ]
        }
      />
      <hr
        data-c-hr="gray"
        data-c-margin="tb(triple)"
      />
      <h2
        data-c-font-size="h2"
        data-c-margin="top(triple) bottom(double)"
      >
        <FormattedMessage
          defaultMessage="All Jobs in"
          description="Message before department name."
          id="hrPortal.jobPageIndex.preDepartmentName"
        />
         Treasury Board of Canada Secretariat
      </h2>
      <p
        data-c-margin="bottom(double)"
      >
        <FormattedMessage
          defaultMessage="This is the list of all active job actions in your department. From here you can 'claim' a job, which will move it into your jobs list above and allow you to begin working with the hiring manager on finding the best talent possible. If you claim a job by accident, no fear, for you can click into the job summary and remove yourself using the 'Relinquish This Job' button."
          description="Message before list of unclaimed jobs."
          id="hrPortal.jobPageIndex.unclaimedJobsMessage"
        />
      </p>
      <section
        data-c-grid="gutter"
      >
        <UnclaimedJobCard
          handleClaimJob={[Function]}
          hiringManager="Amelie Lachance"
          hrAdvisors={Array []}
          id={8}
          jobLink={
            Object {
              "text": "CS03 - Digital Product Designer",
              "title": "",
              "url": "",
            }
          }
          key="8"
          reviewRequested={2020-02-11T06:24:00.000Z}
          status={
            Object {
              "description": Object {
                "en": "This is in review by managers.",
                "fr": "This is in review by HR (FR).",
              },
              "id": 3,
              "key": "review_hr",
              "name": Object {
                "en": "In Review (HR)",
                "fr": "En revue (HR)",
              },
            }
          }
        />
        <UnclaimedJobCard
          handleClaimJob={[Function]}
          hiringManager="Braeden McDoogal"
          hrAdvisors={
            Array [
              "Caitlyn Summers",
              "Jack Little",
            ]
          }
          id={7}
          jobLink={
            Object {
              "text": "FG05 - Long-term Care Specialist",
              "title": "",
              "url": "",
            }
          }
          key="7"
          reviewRequested={2020-02-10T09:24:00.000Z}
          status={
            Object {
              "description": Object {
                "en": "This is a draft.",
                "fr": "This is a draft (FR).",
              },
              "id": 1,
              "key": "draft",
              "name": Object {
                "en": "Draft",
                "fr": "Provisoire",
              },
            }
          }
        />
        <UnclaimedJobCard
          handleClaimJob={[Function]}
          hiringManager="Robin Browne"
          hrAdvisors={Array []}
          id={6}
          jobLink={
            Object {
              "text": "ET03 - Business Analyst",
              "title": "",
              "url": "",
            }
          }
          key="6"
          reviewRequested={2020-02-08T08:24:00.000Z}
          status={
            Object {
              "description": Object {
                "en": "This is in review by managers.",
                "fr": "This is in review by HR (FR).",
              },
              "id": 3,
              "key": "review_hr",
              "name": Object {
                "en": "In Review (HR)",
                "fr": "En revue (HR)",
              },
            }
          }
        />
        <UnclaimedJobCard
          handleClaimJob={[Function]}
          hiringManager="Rebecca Appleby"
          hrAdvisors={
            Array [
              "Rebecca Appleby",
              "Jack Little",
            ]
          }
          id={5}
          jobLink={
            Object {
              "text": "AS02 - Executive Assisstant",
              "title": "",
              "url": "",
            }
          }
          key="5"
          reviewRequested={1995-01-17T10:54:00.000Z}
          status={
            Object {
              "description": Object {
                "en": "This is a draft.",
                "fr": "This is a draft (FR).",
              },
              "id": 1,
              "key": "draft",
              "name": Object {
                "en": "Draft",
                "fr": "Provisoire",
              },
            }
          }
        />
        <UnclaimedJobCard
          handleClaimJob={[Function]}
          hiringManager="Rebecca Appleby"
          hrAdvisors={Array []}
          id={4}
          jobLink={
            Object {
              "text": "CS01 - Front-end Developer",
              "title": "",
              "url": "",
            }
          }
          key="4"
          reviewRequested={1990-12-17T08:24:00.000Z}
          status={
            Object {
              "description": Object {
                "en": "This is a draft.",
                "fr": "This is a draft (FR).",
              },
              "id": 1,
              "key": "draft",
              "name": Object {
                "en": "Draft",
                "fr": "Provisoire",
              },
            }
          }
        />
      </section>
    </div>
  </div>
</section>
`;

exports[`Job Index Page should render JobIndexHr without any items 1`] = `
<section>
  <div
    data-c-background="gray(10)"
    data-c-border="bottom(thin, solid, black)"
  >
    <div
      data-c-container="large"
      data-c-padding="tb(triple)"
    >
      <h2
        data-c-font-size="h2"
        data-c-margin="bottom(normal)"
      >
        <FormattedMessage
          defaultMessage="My Job Actions"
          description="Header for my job actions section."
          id="hrPortal.jobPageIndex.jobActionsHeader"
        />
      </h2>
      <p
        data-c-margin="bottom(double)"
      >
        <FormattedMessage
          defaultMessage="This is a list of all job actions you are currently participating in. Looking for an older job? Check the 'My Completed Job Actions' section below your active jobs."
          description="Message before list of users job actions."
          id="hrPortal.jobPageIndex.jobActionsMessage"
        />
      </p>
      <p>
        <FormattedMessage
          defaultMessage="Claim a job below!"
          description="Message displayed if the jobs actions list is empty."
          id="hrPortal.jobPageIndex.jobActionsEmpty"
        />
      </p>
      <CompletedJobsAccordion
        completedJobActions={Array []}
      />
      <hr
        data-c-hr="gray"
        data-c-margin="tb(triple)"
      />
      <h2
        data-c-font-size="h2"
        data-c-margin="top(triple) bottom(double)"
      >
        <FormattedMessage
          defaultMessage="All Jobs in"
          description="Message before department name."
          id="hrPortal.jobPageIndex.preDepartmentName"
        />
         Treasury Board of Canada Secretariat
      </h2>
      <p
        data-c-margin="bottom(double)"
      >
        <FormattedMessage
          defaultMessage="This is the list of all active job actions in your department. From here you can 'claim' a job, which will move it into your jobs list above and allow you to begin working with the hiring manager on finding the best talent possible. If you claim a job by accident, no fear, for you can click into the job summary and remove yourself using the 'Relinquish This Job' button."
          description="Message before list of unclaimed jobs."
          id="hrPortal.jobPageIndex.unclaimedJobsMessage"
        />
      </p>
      <section
        data-c-grid="gutter"
      />
      <p>
        <FormattedMessage
          defaultMessage="There are currently no active jobs available."
          description="Message displayed if the unclaimed jobs list is empty."
          id="hrPortal.jobPageIndex.unclaimedJobsEmpty"
        />
      </p>
    </div>
  </div>
</section>
`;
